src/lib/moment/get-set.js

```diff
@@ -27,11 +29,27 @@ export function set (mom, unit, value) {
 
 // MOMENTS
 
-export function getSet (units, value) {
-    var unit;
+export function stringGet (units) {
+    units = normalizeUnits(units);
+    if (isFunction(this[units])) {
+        return this[units]();
+    }
+    return this;
+}
+
+
+export function stringSet (units, value) {
     if (typeof units === 'object') {
-        for (unit in units) {
-            this.set(unit, units[unit]);
+        units = normalizeObjectUnits(units);
+        // for (unit in units) {
+        //     this.set(unit, units[unit]);
+        // }
+        var prioritized = getPrioritizedUnits();
+        for (var i = 0; i < prioritized.length; i++)
```

Comment:
What I'd suggest instead, is make a function that takes units and a callback and calls it in the right order. Normally you'd set just a few units, and that will just sort those units and callback with them. In the future if we add 15 calendar systems we'll get hundreds of units, and we have to iterate them every time -- it should be O(number-of-units-to-be-set) not O(number-of-total-units).